src/main/java/sg/ncl/MainController.java

```diff
@@ -3712,6 +3674,30 @@ private String freeTeam(final Team2 team, RedirectAttributes redirectAttributes)
         }
     }
 
+    private void handleException(Team2 team, RedirectAttributes redirectAttributes, MyErrorResource error, ExceptionState exceptionState, String logMessage) {
+        switch (exceptionState) {
+            case TEAM_NOT_FOUND_EXCEPTION:
+                log.warn(logMessage, team.getId(), TEAM_NOT_FOUND);
+                redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + TEAM_NOT_FOUND);
+                break;
+            case INVALID_STATUS_TRANSITION_EXCEPTION:
+                log.warn(logMessage, team.getId(), error.getMessage());
+                redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + error.getMessage());
+                break;
+            case INVALID_TEAM_STATUS_EXCEPTION:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This case's code block is the same as the block for the case on line 3683. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1871)